packages/@aws-cdk/example-construct-library/lib/example-resource.ts

```diff
@@ -0,0 +1,501 @@
+/*
+ * We always import other construct libraries entirely with a prefix -
+ * we never import individual classes from them without a qualifier
+ * (the prefix makes it more obvious where a given dependency comes from,
+ * and prevents conflicting names causing issues).
+ * Our linter also enforces ES6-style imports -
+ * we don't use TypeScript's import a = require('a') imports.
+ */
+import * as cloudformation from '@aws-cdk/aws-cloudformation';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as core from '@aws-cdk/core';
```

Comment:
@eladb once told me that the right way to use this is to import the core classes rather than this way, i.e.,

```ts
import { Construct, Stack } from '@aws-cdk/core'
```

Reply:
I don't like this pattern, and we often don't use it ourselves, don't use it in our docs (like the AWS guide), and don't start with it in the init templates. So I'd rather leave it imported with a qualifier.

Reply:
👍